### PR TITLE
Expose whether we removed encrypted env var to payload

### PR DESCRIPTION
**Pull request by BanzaiMan**
This is necessary so that `travis-build` can display message
about removal

### CODE REVIEW

**Comment by BanzaiMan** [comment]
Some specs are broken, and additional specs are necessary for the new feature.

**Comment by svenfuchs** [comment]
Thanks a lot for doing this, @BanzaiMan! 

Do you need any assistance with the specs?

Reading the changes I was wondering:

* `encrypted_env_removed` as a worker payload key describes the purpose well, but also seems very specific to me. 
* `secure_env` as passed in as an option is more generic (although, now thinking about it, also not quite perfect ... maybe `public_env` would have been a better choice there)

How would you feel about using `secure_env` (or `private_env`, or `public_env`) as a key on the worker payload?

This also might allow removing the extra coupling between `Job::Config` and `Job::Config::Decrypt` since one could just memoize the options hash on `Job::Config` and add an accessor for it on `Job::Config` (then use it in `Payloads::Worker`).

File: lib/travis/scheduler/models/job/config/decrypt.rb [hunk 1]
```diff
@@ -10,6 +10,10 @@ def apply
         config
       end
 
+      def encrypted_env_removed?
+        !options[:secure_env]
```
Comment (BanzaiMan) [inline lib/travis/scheduler/models/job/config/decrypt.rb]: This is actually the exact opposite of `#secure_env?`, which makes it (mostly) superfluous.

What we really need is a convenience method to tell us whether `#remove_encrypted_env` actually removes any variable.

**Comment by BanzaiMan** [comment]
@svenfuchs Yes, I would very much appreciate help on specs.

I've reworked the logic to decouple things apart, but have not completely resolved the name issue.

**Comment by svenfuchs** [comment]
@BanzaiMan I just saw that we already pass `secure_env_enabled` as part of the `job` hash to the worker. https://github.com/travis-ci/travis-scheduler/blob/master/lib/travis/scheduler/payloads/worker.rb#L75

Looking at your new changes it seems that you want to specifically answer the question if any variables have been removed because `secure_env` is `false`. Is that right? Would it be possible to just work with `secure_env_enabled `?

If it's not ... how about renaming the new key to `secure_env_vars_removed` and move it to the `job` hash next to `secure_env_enabled`?

**Comment by BanzaiMan** [comment]
The master branch has progressed quite a bit since this PR was opened. It is no longer feasible to merge from here, so I'll close this and open a new one.

**Pull request closed without merging**